qiskit_aer/quantum_info/states/aer_state.py

```diff
@@ -108,8 +111,13 @@ def configure(self, key, value):
             raise AerError('AerState is configured with a str key')
         if not isinstance(value, str):
             value = str(value)
+        self._configs[key] = value
         self._native_state.configure(key, value)
 
+    def configuration(self):
+        """return configuration"""
+        return self._configs
```

Comment:
This leaks an internal mutable object, where mutating lets things get out of sync (as `self._native_state` won't get updated.  You maybe want to have this be `return self._configs.copy()` to avoid that.